### ISSUE TITLE
using multi-value command-line options

### ISSUE THREAD

**Issue by geddski** [gruntjs/grunt#488]
Grunt uses [nopt](https://npmjs.org/package/nopt) for handling command-line options. `nopt` allows you to pass a set of values assigned to a single option, like so:

`$ node my-program.js --many 1 --many null --many foo`
resulting in
`{ many: ["1", "null", "foo"] }`

But when using `grunt.option("many")` only the last value is returned:
`{ many: "foo" }`

Is there a way to use an array as an options value without adding my own in cli.js?

using grunt 0.4.0a

**Comment by julmot** [comment]
Here is how I solved such an issue:
https://github.com/julmot/jmBoilerplate/commit/0ff15c755f107b1d7b307fbc638022472dfb30dd#diff-35b4a816e0441e6a375cd925af50752cR139
You can pass a list of files to exclude in a task, e.g. `--exclude="index.html, assets/**"`. The one-liner will convert it to an array like `['!index.html', '!assets/**']`.

**Comment by kylezeeuwen** [comment]
@tkellen I'm wondering if a soltuion for this issue ever made it into grunt ? Have done some digging around (haven't looked through source yet) but haven't found anything. I have the same usecase as the OP : `$ node my-program.js --many 1 --many null --many foo` results in `grunt.option('many') = foo`. I can adopt the workaround suggested by @julmot  but wanted to see if there is a built in solution first ? Thanks

**Comment by shama** [comment]
@kylezeeuwen Which version of Grunt are you using? `1.0.1` should allow you to do the following now:

```shell
$ grunt --many=1 --many=null --many=foo
Running "default" task
[ '1', null, 'foo' ]
```

**Comment by kylezeeuwen** [comment]
Hi @shama I am using grunt `0.4.5`. I upgraded to `1.0.1` and can confirm the behaviour you specify above. It appears all my grunt tasks still work which is pretty sweet. 

I did notice that I can no longer specify `grunt task --option foo` as the `=` is now necessary, e.g.,  `grunt task --option=foo` but I imagine this is a known breaking update and deliberate in order to support multi value args with nopts.

Thanks or the quick response I'm satisfied with my service today ;)
